### PR TITLE
Add Authentication modules and basic Authentication Controller

### PR DESCRIPTION
**Pull request by johnkeith**

### CODE REVIEW

**Review by benbutler08 — APPROVED** [review]
This is my current understanding: The Authenticatable module uses our algorithm to give us the encoding and decoding functionality. The Authentication Controller defines a user, based on an email and checks the password. If they match, a token is created by encoding the user. The Authentication module contains the logic that verifies that the current user has a valid token, and assigns it to the right admin. 

Admin includes the Authenticatable module, and Application Controller includes the Authentication module. This gives Admin objects Authenticatable instance methods, and gives everything Authentication instance methods respectively. 

So basically, when a user logs in with a username and password, a token gets created. The user gets matched to the right model object using the token.

**Comment by johnkeith** [comment]
That is right. The Authentication module only applies to controllers. The Authenticatable module only applies to models that we want to allow to authenticate with the system. (That way, we could add more in the future). 

What our front end will do is store that token we create, then use it to make further requests.
